### PR TITLE
feat: amazon-aws - make vault optional

### PR DESCRIPTION
**Pull request by jvm986**
# Description
Add the option to disable authentication with `aws-vault` for users who have vault installed but do not want to use it.

This enables authentication with https://github.com/synfinatic/aws-sso-cli out of the box.

# Screenshot
Screenshot 2023-11-16 at 09 08 23

# Checklist
- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran npm run build and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the assets folder are used by the extension itself
- [x] I checked that assets used by the README are placed outside of the metadata folder

### CODE REVIEW

File: extensions/amazon-aws/src/components/searchbar/aws-profile-dropdown.tsx [hunk 1]
```diff
@@ -1,16 +1,32 @@
...
+  let vaultSessions: string[] | undefined = undefined;
+
+  if (preferences.useAWSVault) {
+    vaultSessions = useVaultSessions();
+    isUsingAwsVault = !!vaultSessions;
+    useAwsVault({
+      profile: vaultSessions?.includes(selectedProfile || "") ? selectedProfile : undefined,
+      onUpdate: () => onProfileSelected?.(),
+    });
+  }
```
Comment (JonathanWbn) [inline extensions/amazon-aws/src/components/searchbar/aws-profile-dropdown.tsx:28]: Calling hooks conditionally is [not allowed](https://legacy.reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level) and might lead to unexpected behavior.

Instead, I suggest using the `execute` config field of the `useExec` call in the `useAwsVault` hook. If `execute` is `false` in that hook, this should be an effective way to disable aws-vault.

**Review by JonathanWbn — COMMENTED** [review]
Nice! 👍
I left a comment with a suggestion on how else to disable aws-vault.

**Comment by pernielsentikaer** [comment]
Could you please tag @JonathanWbn and me when you're ready for another review 🙂

**Review by JonathanWbn — APPROVED** [review]

File: extensions/amazon-aws/src/components/searchbar/aws-profile-dropdown.tsx (hunk 1 above)
Comment (jvm986) [reply extensions/amazon-aws/src/components/searchbar/aws-profile-dropdown.tsx]: fixed, thanks

**Comment by jvm986** [comment]
@pernielsentikaer I think this is good to go now

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**